index.ts

```diff
@@ -221,18 +221,31 @@ class Applesign {
     await tools.asyncRimraf(placeholderdir);
   }
 
-  // XXX some directory leftovers
   async removeXCTests() {
     fchk(arguments, []);
+    let dirsToCheck: Set<string> = new Set<string>();
     const dir = this.config.appdir;
     walk.walkSync(dir, (basedir: string, filename: string, stat: any) => {
       const target = path.join(basedir, filename);
       //  if (target.toLowerCase().indexOf('/xct') !== -1)
       if (target.toLowerCase().indexOf("xctest") !== -1) {
```

Comment:
```suggestion
      if (target.toLowerCase().indexOf(".xctest") !== -1) {
```

This check is probably safer